ci_scripts/tag_and_release.py

```diff
@@ -1,107 +1,129 @@
-    version = subprocess.check_output(['python', 'setup.py', '--version']).decode().strip()
-
-    twine_repo = os.getenv('TWINE_REPOSITORY_URL') or os.getenv('TWINE_REPOSITORY')
-    print('tagging and releasing %r as a %s release to %s as Twine user %s' % (
-        version,
-        mode.name,
-        twine_repo,
-        os.getenv('TWINE_USERNAME')
-    ))
-
-    print('python - preparing environment')
-    subprocess.check_call(['apk', 'update'])
-    subprocess.check_call(['apk', 'add', 'git'])
-    # Pin twine to 1.11 due to https://github.com/pypa/twine/issues/404
-    subprocess.check_call(['pip', 'install', 'twine<1.12'])
-    url = subprocess.check_output(['git', 'remote', 'get-url', 'origin'])
-    new_url = git_url_ssh_to_https(url.decode())
-    subprocess.check_call(['git', 'remote', 'set-url', 'origin', new_url])
-    branch_spec = 'origin/%s' % os.getenv('CIRCLE_BRANCH')
-    subprocess.check_call(['git', 'branch', '--set-upstream-to', branch_spec])
-    subprocess.check_call(['git', 'fetch', '--tags', '--force'])
-
-    print('git - add changes')
-    subprocess.check_call(['git', 'add', 'src/mbed_cloud/_version.py'])
-    subprocess.check_call(['git', 'add', 'CHANGELOG.rst'])
-    subprocess.check_call(['git', 'add', 'docs/news/*'])
-    message = ':checkered_flag: :newspaper: releasing version %s\n[skip ci]' % version
-    subprocess.check_call(['git', 'commit', '-m', message])
-
-    if mode == release_target_map['prod']:
-        print('git - pushing %s changelog commit' % mode.name)
-        subprocess.check_call(['git', 'push', 'origin'])
-
-    # tags
-    subprocess.check_call(['git', 'tag', '-a', version, '-m', 'release %s' % version])
-    subprocess.check_call(['git', 'tag', '-f', 'latest'])
-    if mode == release_target_map['prod']:
-        print('git - pushing %s tags' % mode.name)
-        subprocess.check_call(['git', 'push', '-f', 'origin', '--tags'])
-
-    print('pypi - uploading')
-    subprocess.check_call(['python', '-m', 'twine', 'upload', mode.bundle])
-    print('pypi - uploading successful')
+    twine_repo = os.getenv(f'{ENVVAR_TWINE_REPOSITORY_URL}') or os.getenv(
+        ENVVAR_TWINE_REPOSITORY)
+    twine_username = os.getenv(ENVVAR_TWINE_USERNAME)
+
+    if not gh_token:
+        raise ValueError(
+            f'Environment variable [{ENVVAR_GIT_TOKEN}] (github token) is not set.'
+        )
+
+    if not twine_repo or not twine_username:
+        error_message = f'Environment variables [{ENVVAR_TWINE_REPOSITORY}/{ENVVAR_TWINE_REPOSITORY_URL}] (PyPI repository/URL)'
+        + f' and/or [{ENVVAR_TWINE_USERNAME}] (PyPI username) are not set.'
+        raise ValueError(error_message)
+
+
+def _release_to_pypi():
+    logger.info('Releasing to PyPI')
+    logger.info('Generating a release package')
+    with cd(PROJECT_ROOT):
+        subprocess.check_call(
+            [sys.executable, 'setup.py', 'clean', '--all',
+             'bdist_wheel',
```

Comment:
How have you decided on using `bdist_wheel`? I don't know enough about how this works, do we not want to include source as well (sdist)?

Reply:
I used whatever command was there before but I agree `sdist` option is missing.
We should follow what is said in https://packaging.python.org/tutorials/packaging-projects/